### ISSUE TITLE
Do not download source bundle if cache contains bundle

### ISSUE THREAD

**Issue by aronatkins** [rstudio/packrat#221]
Packrat should avoid downloading package bundles under situations where it knows that the package is already available in the packrat cache. 

It might be safe to make this change only after packrat stores full DESCRIPTION file content.

This is most visible when building multiple applications in a low-bandwidth environment.

**Comment by kevinushey** [comment]
This highlights a growing tension in packrat -- it was originally developed purely for _reproducible_ project management, and so having the package sources for whatever your project was using was a primary goal.

Now, we're trying to make packrat do more -- reproducible should be the default, but we should be able to 'bend' the rules when desired. @yihui posted a similar-in-scope request on the mailing list (https://groups.google.com/forum/#!topic/packrat-discuss/nmvW6EopDLQ).

Anyway, for this, I think a project option of the form 'keep.package.sources' would be appropriate here.

**Comment by aronatkins** [comment]
Another option: store package sources as part of the packrat cache instead of within each directory. Treat it as a front for the authoritative source.

**Comment by donarus** [comment]
I agree with @aronatkins proposal - it sounds good to store sources the same way (cache) as installed packages.
